tokio/docs/runtime-stats.md

```diff
@@ -0,0 +1,137 @@
+
+Some of the stats include min/max (specifically the queue depth stats) this is because the depth of the queues changes throughout the stats batch window. The value could start low, spike up during the middle of the window then come back down. To understand this behavior the executor stats module will aggregate the depth values to reduce the need to stream the values.
+
+### Global queue
+
+Since, the global inject queue is not attached to a single worker its length value can be sampled directly from collector thread. In addition, the global inject queue length is already tracked via an `AtomicUsize`, therefore this value can just be exposed via the `Stats` struct directly.
+
+Stats exposed for the global inject queue:
+- Queue length
+
+### Blocking operations
+
+The blocking pool already tracks the number of idle threads and the total number of threads. These values are currently within a shared mutex but can be moved to be `AtomicUsize` values and then shared with the `Stats` struct to be sampled by the collector. In addition, a counter that is incremented on each task execution will be included. All values will be streamed to the stats struct via atomics.
+
+Stats from the blocking pool:
+- Number of idle threads
+- Number of threads
+- Number of blocking task
+- Blocking task queue depth
+
+### Task
+
+This implementation will avoid tracking stats/metrics at the task level due to the overhead required. This will instead be accomplished by the [tokio console](https://github.com/tokio-rs/console). This will allow the user to attach the console and take the performance hit when they want to explore issues in more detail.
+
+### I/O driver
+
+Unlike, the executor stats, stats coming from the I/O driver will be streamed directly to the `Stats` struct via atomics. Each value will be incremented (via `AtomicU64::fetch_add`) for each event.
+
+List of stats provided from the I/O driver:
+- Amount of compact
+- Amount of "token dispatches" (aka ready events)
+- Amount of fd currently registered with `io::Driver`
+- Amount of read ready events
+- Amount of write ready events
+
+### Timer driver
+
+Similar to the I/O driver the timer driver will stream stats in a similar fashion. These will be emitted via `AtomicU64::fetch_add` and be emitted on each occurance of the events below.
+
+List of stats provided from the timer driver:
+- Amount of timer entries (timer wheel depth)
+- Amount of timers fired
+- Amount of timers cancelled
+
+### `tokio-metrics`
+
+The `tokio-metrics` crate will provide aggregated metrics based on the `Stats` struct. This will include histograms and other useful aggregated forms of the stats that could be emitted by various metrics implementations. This crate is designed to provide the ability to expose the aggregated stats in an unstable `0.1` method outside of the runtime and allow the ability to iterate on how they are aggregated without the need to follow `tokio`'s strict versioning scheme.
```

Comment:
It would help to expand on this. What specific aggregations will tokio-metrics expose. How do you expect this to be used in practice? What will trigger alerts, how are engineers expected to use the aggregations in their workflow, etc...